### PR TITLE
feat: create ePortfolio notification

### PR DESCRIPTION
**Pull request by Judge40**
The e-portfolio notification should be used for new starters taking part
in the pilot.
It will only be created, as an in-app notification if it has not been
created before.

TIS21-5615

### CODE REVIEW

**Review by ReubenRobertsHEE — APPROVED** [review]
Looks good. I assume you're ok with the whitelist/staging logic missing for now - given that we've disabled the lambda we don't need to worry about the DMS at this point,  and the new logic to configure send vs. log-only for in-app and email notifications should be merged in today (or this week anyway!)

File: src/main/java/uk/nhs/tis/trainee/notifications/service/NotificationService.java [hunk 1]
```diff
@@ -288,6 +289,44 @@ private UserDetails getTraineeDetails(String personId) {
...
+   * @return true if all criteria bet, or false if one or more criteria fail.
+   */
+  public boolean meetsCriteria(ProgrammeMembership programmeMembership,
+      boolean checkNewStarter, boolean checkPilot) {
+    String traineeId = programmeMembership.getPersonId();
+    String pmId = programmeMembership.getTisId();
+
+    if (checkNewStarter) {
+      String apiPath = "/api/programme-membership/isnewstarter/{traineeId}/{pmId}";
+      Boolean isNewStarter = restTemplate.getForObject(serviceUrl + apiPath, Boolean.class,
```
Comment (ReubenRobertsHEE) [inline src/main/java/uk/nhs/tis/trainee/notifications/service/NotificationService.java:307]: If the API call fails (e.g. if trainee-details is down), I presume not wrapping this in a try-catch will mean the exception that is thrown will requeue the original message. In my branch I was returning a default 'false', but I think I prefer this behaviour.

Comment (Judge40) [reply src/main/java/uk/nhs/tis/trainee/notifications/service/NotificationService.java:307]: 🤔 I'll have to double check that

Comment (Judge40) [reply src/main/java/uk/nhs/tis/trainee/notifications/service/NotificationService.java:307]: I think I misunderstood your comment originally. I thought you were seeing it default to false and discarding the message when the details service was down 😬

Comment (ReubenRobertsHEE) [reply src/main/java/uk/nhs/tis/trainee/notifications/service/NotificationService.java:307]: Ah no, sorry - bad phrasing. You're right: if the details service is down then it should requeue the message. The work I was doing in parallel had been trapping any REST errors and discarding the message, but I'm going to amend it to match your code.

**Pull request merged**